draft-thomson-http-mice.md

```diff
@@ -264,62 +266,50 @@ status code.  However, if the integrity proof for the first record is not known,
 this check SHOULD NOT fail unless explicitly configured to do so.
 
 
-# The MI HTTP Header Field {#header}
+# The mi-sha256 digest algorithm {#digest-mi-sha256}
 
-The MI HTTP header field carries message integrity proofs corresponding to
-content encoding(s) that have been applied to a payload body.
+[RFC3230] describes digests applying to "the entire instance associated with the
+message". The instance corresponds to the "representation" in Section 3 of
+[RFC7231], but unlike the existing digest algorithms, the mi-sha256 digest
+algorithm specifies the top-level digest at the point when the mi-sha256 content
+coding ({{encoding}}) is applied or removed from the representation.
 
-The MI header field uses the extended ABNF syntax defined in Section 1.2 of
-[RFC7230] and the `parameter` rule from [RFC7231]:
+When the "mi-sha256" digest algorithm is specified for a representation, the
+recipient MUST:
```

Comment:
This construction is problematic.  It says that you have to follow a process, rather than describing the semantics.  It also over-constrains implementations by mandating everything in the list (via this "MUST").  This could easily be prose.

Reply:
Ah, I'm used to working on specs that include permission like ["Conformance requirements phrased as algorithms or specific steps may be implemented in any manner, so long as the end result is equivalent."](https://infra.spec.whatwg.org/#algorithms), and I missed that IETF ones don't have it. I've updated this section to avoid needing that.